src/agent.spec.ts

```diff
@@ -1,74 +1,221 @@
-        Finding.fromObject({
-          name: "High Tether Transfer",
-          description: `High amount of USDT transferred: ${normalizedValue}`,
-          alertId: "FORTA-1",
-          severity: FindingSeverity.Low,
+        expect.objectContaining({
+          name: "Nethermind Bot Deployment Detector",
+          description: "Detects Bots Deployed by Nethermind",
+          alertId: "FORTA-123",
+          severity: FindingSeverity.Info,
+          type: FindingType.Info,
+          metadata: {
+            agentId: mockTxEventOne.agentId.toString(),
+            owner: mockTxEventOne.owner,
+            metadata: mockTxEventOne.metadata,
+            chainIds: mockTxEventOne.chainIds[0],
+          },
+        }),
+        expect.objectContaining({
+          name: "Nethermind Bot Deployment Detector",
+          description: "Detects Bots Deployed by Nethermind",
+          alertId: "FORTA-123",
+          severity: FindingSeverity.Info,
           type: FindingType.Info,
           metadata: {
-            to: mockTetherTransferEvent.args.to,
-            from: mockTetherTransferEvent.args.from,
+            agentId: mockTxEventTwo.agentId.toString(),
+            owner: mockTxEventTwo.owner,
+            metadata: mockTxEventTwo.metadata,
+            chainIds: mockTxEventTwo.chainIds[0],
           },
         }),
       ]);
-      expect(mockTxEvent.filterLog).toHaveBeenCalledTimes(1);
-      expect(mockTxEvent.filterLog).toHaveBeenCalledWith(
-        ERC20_TRANSFER_EVENT,
-        TETHER_ADDRESS
+    });
+    it("returns empty findings if there is a bot deployment with the wrong function abi", async () => {
+      const newProxyInterface = new Interface([INCORRECT_FUNCTION_SIGNATURE]);
+
+      mockTxEvent = new TestTransactionEvent()
+        .setFrom(BOT_DEPLOYER_ADDRESS)
+        .setTo(FORTA_BOT_REGISTRY)
+        .addTraces({
+          function: newProxyInterface.getFunction("fooAgent"),
+          to: FORTA_BOT_REGISTRY,
+          from: BOT_DEPLOYER_ADDRESS,
+          arguments: [
+            mockTxEventOne.agentId,
+            mockTxEventOne.owner,
+            mockTxEventOne.metadata,
+            [BigNumber.from(mockTxEventOne.chainIds[0])],
+          ],
+        });
+
+      findings = await handleTransaction(mockTxEvent);
+
+      //make sure findings is equal to an empty array
+      expect(findings).toStrictEqual([]);
+    });
+    it("returns empty findings if there is a bot deployment with a call to newAgent function NOT in the fortaContract", async () => {
+      const newProxyInterface = new Interface([EXTERNAL_AGENT_FUNCTION_SIGNATURE]);
+      // call provideHandle Transaction with external function ABI
+      handleTransaction = provideHandleTransaction(
+        FORTA_BOT_REGISTRY,
+        BOT_DEPLOYER_ADDRESS,
+        EXTERNAL_AGENT_FUNCTION_SIGNATURE
       );
+      let mockTxEvent = new TestTransactionEvent()
+        .setFrom(BOT_DEPLOYER_ADDRESS)
+        .setTo(FORTA_BOT_REGISTRY)
```

Comment:
I may not have explained myself well before. This test would be pretty much the same as `"returns empty findings if the deployer is NOT Nethermind"`, just passing a non-`FORTA_BOT_REGISTRY` address to `.setTo()`. So it would be the same function ABI, just calling the incorrect contract.

If I said "external" previously, I didn't mean the function's visibility, but rather a contract external to the Forta ecosystem of contracts.

With that change, you wouldn't need that second `if` statement in the bot logic, nor to create the `EXTERNAL_AGENT_FUNCTION_SIGNATURE`.

Hope that clarifies some things.